brayns/common/scene/Scene.cpp

```diff
@@ -984,158 +490,317 @@ void Scene::loadFromCacheFile()
+        material->setRefractionIndex(1.5f);
+        material->setReflectionIndex(0.1f);
+        material->setDiffuseColor(WHITE);
+        material->setSpecularColor(WHITE);
+        material->setSpecularExponent(100.f);
+        model->addSphere(materialId, {{0.25f, 0.26f, 0.30f}, 0.25f});
+        ++materialId;
+    }
+
+    {
+        // Cylinder
+        auto material = model->createMaterial(materialId, "cylinder");
+        material->setDiffuseColor({0.1f, 0.1f, 0.8f});
+        material->setSpecularColor(WHITE);
+        material->setSpecularExponent(10.f);
+        model->addCylinder(materialId, {{0.25f, 0.126f, 0.75f},
+                                        {0.75f, 0.126f, 0.75f},
+                                        0.125f});
+        ++materialId;
+    }
+
+    {
+        // Cone
+        auto material = model->createMaterial(materialId, "cone");
+        material->setReflectionIndex(0.8f);
+        material->setSpecularColor(WHITE);
+        material->setSpecularExponent(10.f);
+        model->addCone(materialId, {{0.75f, 0.01f, 0.25f},
+                                    {0.75f, 0.5f, 0.25f},
+                                    0.15f,
+                                    0.f});
+        ++materialId;
+    }
+
+    {
+        // Lamp
+        auto material = model->createMaterial(materialId, "lamp");
+        material->setDiffuseColor(WHITE);
+        material->setEmission(5.f);
+        const Vector3f lampInfo = {0.15f, 0.99f, 0.15f};
+        const Vector3f lampPositions[4] = {
+            {0.5f - lampInfo.x(), lampInfo.y(), 0.5f - lampInfo.z()},
+            {0.5f + lampInfo.x(), lampInfo.y(), 0.5f - lampInfo.z()},
+            {0.5f + lampInfo.x(), lampInfo.y(), 0.5f + lampInfo.z()},
+            {0.5f - lampInfo.x(), lampInfo.y(), 0.5f + lampInfo.z()}};
+        auto& trianglesMesh = model->getTrianglesMeshes()[materialId];
+        for (size_t i = 0; i < 4; ++i)
+            trianglesMesh.vertices.push_back(lampPositions[i]);
+        trianglesMesh.indices.push_back(Vector3i(2, 1, 0));
+        trianglesMesh.indices.push_back(Vector3i(0, 3, 2));
+    }
+}
+
+void Scene::setMaterialsColorMap(MaterialsColorMap colorMap)
 {
-    _materials.push_back(material);
-    return _materials.size() - 1;
+    for (auto& modelDescriptors : _modelDescriptors)
+        modelDescriptors.getModel()->setMaterialsColorMap(colorMap);
 }
 
-void Scene::setMaterial(const size_t index, const Material& material)
+Boxf Scene::getBounds()
 {
-    _buildMissingMaterials(index);
-    _materials[index] = material;
+    size_t nbEnabledModels{0};
+    Boxf bounds;
+    for (const auto& modelDescriptor : _modelDescriptors)
+        if (modelDescriptor.getEnabled())
+        {
+            bounds.merge(modelDescriptor.getModel()->getBounds());
+            ++nbEnabledModels;
+        }
+    if (nbEnabledModels == 0)
+    {
+        // If no model is enabled. return unity bounding box
+        bounds.merge({0, 0, 0});
```

Comment:
why? if nothing is visible, there is actually not box :) Is this needed somewhere not to be empty?